code/game/gamemodes/miniantags/abduction/abduction_gear.dm

```diff
@@ -337,17 +337,21 @@ CONTENTS:
 
 	add_attack_logs(user, L, "Stunned with [src]")
 
-/obj/item/abductor_baton/proc/SleepAttack(mob/living/L,mob/living/user)
-	if(L.IsStunned() || L.IsSleeping())
-		L.visible_message("<span class='danger'>[user] has induced sleep in [L] with [src]!</span>", \
-							"<span class='userdanger'>You suddenly feel very drowsy!</span>")
-		playsound(loc, 'sound/weapons/egloves.ogg', 50, TRUE, -1)
-		L.Sleeping(120 SECONDS)
-		add_attack_logs(user, L, "Put to sleep with [src]")
-	else
-		L.AdjustDrowsy(2 SECONDS)
-		to_chat(user, "<span class='warning'>Sleep inducement works fully only on stunned specimens!</span>")
-		L.visible_message("<span class='danger'>[user] tried to induce sleep in [L] with [src]!</span>", \
+/obj/item/abductor_baton/proc/SleepAttack(mob/living/L, mob/living/user)
+	if(!iscarbon(L))
+		return
+	var/mob/living/carbon/C = L
+	if(do_mob(user, C, 2.5 SECONDS))
+		if(C.getStaminaLoss() > 100 || C.IsSleeping())
```

Comment:
Make these 2 into guard clauses please
if(!do_mob())
    return
 if(not sleepable)
     do_drowsy_code()
     return
  force them to sleep

Reply:
thoughts on changing the proc to read as such?
```	/obj/item/abductor_baton/proc/SleepAttack(mob/living/L, mob/living/user)
	var/mob/living/carbon/C = L
	if(!iscarbon(L))
		return
	if(C.IsSleeping())
		C.visible_message("<span class='notice'>[C] is already asleep!</span>")
		return
	if(C.getStaminaLoss() < 100)
		C.AdjustDrowsy(2 SECONDS)
		to_chat(user, "<span class='warning'>Sleep inducement works fully only on stunned specimens!</span>")
		C.visible_message("<span class='danger'>[user] tried to induce sleep in [L] with [src]!</span>", \
						"<span class='userdanger'>You suddenly feel drowsy!</span>")
		return
	if(do_mob(user, C, 2.5 SECONDS))
		C.visible_message("<span class='danger'>[user] has induced sleep in [L] with [src]!</span>", \
							"<span class='userdanger'>You suddenly feel very drowsy!</span>")
		playsound(loc, 'sound/weapons/egloves.ogg', 50, TRUE, -1)
		C.Sleeping(120 SECONDS)
		add_attack_logs(user, C, "Put to sleep with [src]")
	else
		return
```

Reply:
yes, but you need to invert the `if(C.IsSleeping())` clause to `if(!C.IsSleeping())` to keep the same logic